pilot/pkg/security/authn/v1beta1/policy_applier_test.go

```diff
@@ -363,6 +363,112 @@ func TestJwtFilter(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "Forward original token",
+			in: []*model.Config{
+				{
+					Spec: &v1beta1.RequestAuthentication{
+						JwtRules: []*v1beta1.JWTRule{
+							{
+								Issuer:               "https://secret.foo.com",
+								JwksUri:              jwksURI,
+								ForwardOriginalToken: true,
+							},
+						},
+					},
+				},
+			},
+			expected: &http_conn.HttpFilter{
+				Name: "envoy.filters.http.jwt_authn",
+				ConfigType: &http_conn.HttpFilter_TypedConfig{
+					TypedConfig: pilotutil.MessageToAny(
+						&envoy_jwt.JwtAuthentication{
+							Rules: []*envoy_jwt.RequirementRule{
+								{
+									Match: &route.RouteMatch{
+										PathSpecifier: &route.RouteMatch_Prefix{
+											Prefix: "/",
+										},
+									},
+									Requires: &envoy_jwt.JwtRequirement{
+										RequiresType: &envoy_jwt.JwtRequirement_AllowMissingOrFailed{
+											AllowMissingOrFailed: &empty.Empty{},
+										},
+									},
+								},
+							},
+							Providers: map[string]*envoy_jwt.JwtProvider{
+								"origins-0": {
+									Issuer: "https://secret.foo.com",
+									JwksSourceSpecifier: &envoy_jwt.JwtProvider_LocalJwks{
+										LocalJwks: &core.DataSource{
+											Specifier: &core.DataSource_InlineString{
+												InlineString: test.JwtPubKey1,
+											},
+										},
+									},
+									Forward:           true,
+									PayloadInMetadata: "https://secret.foo.com",
+								},
+							},
+						}),
+				},
+			},
+		},
+		{
+			name: "Output payload",
+			in: []*model.Config{
+				{
+					Spec: &v1beta1.RequestAuthentication{
+						JwtRules: []*v1beta1.JWTRule{
+							{
+								Issuer:                "https://secret.foo.com",
+								JwksUri:               jwksURI,
+								ForwardOriginalToken:  true,
+								OutputPayloadToHeader: "x-foo",
```

Comment:
what's the behavior if two rules specify the same payload to header? the last wins? should we add a test for that?

Reply:
Yes. On the other hand, we do not support multiple JWTs per request (yet). We will need to document it clearly in istio.io, but for now, I don't think that test case is necessary.